client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,389 @@
+             name
+	         members {
+		       costumeColor
+             }
+         }
+  }
+}
+----
+Note that there are 2 arguments on the addHeroToTeam field. The way to run it using the proposed API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.MUTATION)
+       .addRootField("addHeroToTeam",
+               asSet(
+                       arg("hero", "Wolverine"),
+                       arg("team", "Avengers")
+               ),
+               asSet(
+                       field("name"),
+                       field("rivalTeam",
+                               asSet(
+                                       field("name"),
+                                       field("members",
+                                               asSet(
+                                                       field("costumeColor")
+                                               )
+                                       )
+                               )
+                       )
+               )
+       )
+
+----
+
+== Managing input objects
+
+In that case, we want to create a hero from scratch:
+[source,json]
+----
+mutation {
+  createNewHero(hero: {
+	name: "Green Lantern",
+    	costumeColor: "green",
+    	dateOfLastCheckin: "2020-03-25",
+    	equipment: null,
+    	knownEnemies: [“Wolverine, “Starlord”, Iron Man”],
+	idNumber: 456) {
+    realName
+  }
+}
+----
+We can observe that we have a multi-valued argument and we need a specific API construct for that: the InputObject.For instance:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.MUTATION)
+       .addRootField(field("createNewHero",
+               args(
+                       arg("hero", inputObject(
+                                   inputField("costumeColor", "green"),
+                                   inputField("dateOfLastCheckin", LocalDate.now()),
+                                   inputField("equipment", null),
+                                   inputField("idNumber", 456),
+                                   inputField("knownEnemies", asList("Wolverine", "Starlord", "Iron Man")),
+                                   inputField("name", "Green Lantern")
+                       ))
+               ),
+               fields(
+                       field("name"),
+                       field("nameOfKnownEnemies")
+               )
+       ));
+----
+
+== Running operations
+
+Once the operation is built (a mutation or a query), it is ready to be run against an endpoint. Let's see how to do that.
+
+There are two options depending on the way to get the data result.
+
+To get raw JSON data:
+[source,Java]
+----
+// Build the request
+Request request = builder.build();
+
+// Initialize the GraphQL client
+Client client = ClientBuilder.newClient();
+
+// Get the GraphQL response
+Response response = client.target(“http://localhost:8080/graphql”).execute(request);
+
+// Check GraphQL errors if any
+if ( response.hasError ) {
```

Comment:
Should the `hasError` member be a field or a method?  I tend to favor a method, but it doesn't have to be.  Wdyt?

Reply:
I'd also prefer a method, I think.

Reply:
You're right, it is a typo. To be fixed.